host/xtest/xtest_6000.c

```diff
@@ -856,6 +1015,169 @@ static void xtest_tee_test_6009(ADBG_Case_t *c)
 	TEEC_CloseSession(&sess);
 }
 
+#ifdef CFG_GP_TESTSUITE_ENABLE
+static void xtest_tee_test_6010(ADBG_Case_t *c)
+{
+    TEEC_Session sess;
+    uint32_t orig;
+    uint8_t out[4000] = {0};
+    uint8_t in[0x12c] = {'b'};
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, xtest_teec_open_session(
+                &sess, &gp_tta_ds_uuid, NULL, &orig)))
+        return;
+
+    Do_ADBG_BeginSubCase(
+        c, "GP DS CreatePersistentObject AccessConflict (9d-1d-62)");
+
+    if (!ADBG_EXPECT_TEEC_RESULT(
+            c, TEEC_ERROR_ACCESS_CONFLICT, ds_open_access_conf(&sess)))
+        goto exit;
+
+    Do_ADBG_EndSubCase(
+        c, "GP DS CreatePersistentObject AccessConflict (9d-1d-62)");
+    Do_ADBG_BeginSubCase(
+        c, "GP DS RestrictObjectUsagePanic (9d-5d-46)");
+
+    if (!ADBG_EXPECT_TEEC_RESULT(
+            c, TEE_ERROR_TARGET_DEAD, ds_res_obj_panic(&sess)))
+        goto exit;
+
+    TEEC_CloseSession(&sess);
+
+    Do_ADBG_EndSubCase(
+        c, "GP DS RestrictObjectUsagePanic (9d-5d-46)");
+    Do_ADBG_BeginSubCase(
+        c, "GP DS SeekObjectData BadHandle (9d-c3-c8)");
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, xtest_teec_open_session(
+                &sess, &gp_tta_ds_uuid, NULL, &orig)))
+        return;
+
+    if (!ADBG_EXPECT_TEEC_RESULT(
+            c, TEE_ERROR_TARGET_DEAD, ds_seek_obj_bad_handle(&sess)))
+        goto exit;
+
+    TEEC_CloseSession(&sess);
+
+    Do_ADBG_EndSubCase(
+        c, "GP DS SeekObjectData BadHandle (9d-c3-c8)");
+    Do_ADBG_BeginSubCase(
+        c, "GP DS SeekObjectData NotPersist (9d-db-4a)");
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, xtest_teec_open_session(
+                &sess, &gp_tta_ds_uuid, NULL, &orig)))
+        return;
+
+    if (!ADBG_EXPECT_TEEC_RESULT(
+            c, TEE_ERROR_TARGET_DEAD, ds_seek_obj_inv_handle(&sess)))
+        goto exit;
+
+    TEEC_CloseSession(&sess);
+
+    Do_ADBG_EndSubCase(
+        c, "GP DS SeekObjectData NotPersist (9d-db-4a)");
+    Do_ADBG_BeginSubCase(c, "GP DS SeekWriteRead SEEK_END (9d-e4-58)");
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, xtest_teec_open_session(
+                &sess, &gp_tta_ds_uuid, NULL, &orig)))
+        return;
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, ds_seek_gp(
+                &sess, TEE_DATA_SEEK_END, 0, 2, data_00, sizeof(data_00), out,
+                sizeof(out))))
+        goto exit;
+
+    /* check buffer */
+    (void)ADBG_EXPECT_BUFFER(
+        c, data_00, sizeof(data_00), out, sizeof(data_00));
+
+    if (!ADBG_EXPECT_TEEC_SUCCESS(
+            c, ds_seek_gp(
+                &sess, TEE_DATA_SEEK_END, sizeof(in), 0, in, sizeof(in), out,
+                sizeof(out))))
+        goto exit;
+
+    (void)ADBG_EXPECT_BUFFER(c, in + sizeof(in), sizeof(in), out, sizeof(in));
```

Comment:
Can we pass the test with newest optee_os? It looks like has the same issue @jforissier mentioned before in the mail.